docs/adrs/0002-handling-migration-from-sfdx-to-sf-cli-in-cumulusci.md

```diff
@@ -0,0 +1,75 @@
+    - Bad: There could be problems with the new SF CLI we don't know about yet.
+
+3. **Let SF CLI handle old SFDX commands:**
+    - Good: Reduces changes to our commands, making the switch smoother for users.
+    - Good: Lets us switch to the new CLI slowly, reducing risk of big issues.
+    - Bad: Delays the inevitable. If they drop deprecated commands without us noticing, it would break functionality for our users.
+    - Bad: Could delay us learning and using new features of the SF CLI.
+
+### Decision Outcome
+
+Here's how we plan to make this change:
+
+-   **Maintain Backwards Compatibility:** Our priority is to ensure that all existing processes continue to function as expected. We will not modify the existing SFDX commands in our project until SF CLI announces the version in which these commands will be deprecated.
+
+-   **Delegate maintaining backwards compatibility to the SF CLI:** We will continue using existing commands as written and rely on SF CLI to interpret these commands. We think this will make the change easier and less disruptive.
+
+-   **Warn users that are on SFDX (v7) that they need to upgrade:** We'll check the installed version of the CLI and warn users that future versions of CumulusCI will drop support.
+
+-   **Update our integration tests to include `stable-rc`:** We'll revisit PR #3558 to make sure all tasks that use the CLI are tested in our workflow, and that we're testing the weekly release candidate channel (`stable-rc`).
+
+-   **Update the Falcon web applications to depend on SF as soon as practical:** We know we need to switch to SF CLI, and we want our Falcon web apps to benefit from its latest updates and features as soon as possible. We'll also install all necessary JIT plugins in the apps, so we don't have to install them over and over during builds.
+
+-   **Update the SFDX Heroku buildpack to depend on SF as soon as practical:** Like the Falcon web applications, we want our [Heroku buildpack](https://github.com/SalesforceFoundation/simple-salesforce-dx-buildpack)to stay up-to-date with the latest SF CLI to get its performance and feature enhancements. Like the Falcon web applications, this includes installing all necessary JIT plugins for the buildpack.
+
+## Consequences
+
+-   Keeping things working as they are will cause the least disruption.
+-   By delegating backwards compatibility to the SF CLI, we can avoid making substantial changes to our command structure until it's necessary.
+-   Updating Falcon web apps and the SFDX Heroku buildpack to use the SF CLI will help us stay ahead of the change and avoid a scramble as the CLI removes deprecated commands.
+-   Installing JIT plugins in the apps and buildpack will make builds more efficient and avoid problems during runtime.
+-   However, this decision does mean that we will need to monitor Salesforce's communications closely for announcements about command deprecations, to ensure that our commands continue to function as expected.
```

Comment:
😬.

Our use of SFDX is really not wide. I wonder if we could just do an inventory and see if there are `sf` commands that map 1:1 with the same semantics.

There might be some stickyness around user-created tasks that use the SFDXCommandTask class, but I can see a possibility that the backwards-compatibility you mention above might be enough to address that. We could also log warnings to the user from that class.

Reply:
> 😬 

Yeah.

The catch is that the `sf` CLI may or may not be present, depending on the installation method: The `pkg` and other installers bundle the `sfdx` and `sf` CLIs, while NPM-based installations do not. Even if the `sf` CLI is available, the v2 SF commands that are the direct equivalents of the deprecated SFDX commands we invoke may not be available if it's v1.

In comparison, SFDXCommandTask-based tasks can remain unchanged, since v2 SF commands work with the `sfdx` alias.